test/unit/TicketBroker.js

```diff
@@ -620,4 +648,245 @@ contract("TicketBroker", accounts => {
             assert.equal(events[0].returnValues.recipient, recipient)
         })
     })
+
+    describe("unlock", () => {
+        it("reverts when both deposit and penaltyEscrow are zero", async () => {
+            await expectRevertWithReason(broker.unlock(), "sender deposit and penalty escrow are zero")
+        })
+
+        it("reverts when called twice", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.unlock()
+
+            await expectRevertWithReason(broker.unlock(), "unlock already initiated")
+        })
+
+        it("reverts when called twice by multiple senders", async () => {
+            const sender2 = accounts[2]
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.fundDeposit({from: sender2, value: 2000})
+            await broker.unlock({from: sender})
+            await broker.unlock({from: sender2})
+
+            await expectRevertWithReason(broker.unlock({from: sender}), "unlock already initiated")
+            await expectRevertWithReason(broker.unlock({from: sender2}), "unlock already initiated")
+        })
+
+        it("sets withdrawBlock according to constructor config", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            const fromBlock = (await web3.eth.getBlock("latest")).number
+
+            await broker.unlock({from: sender})
+
+            const withdrawBlock = (await broker.senders.call(sender)).withdrawBlock.toString()
+            const expectedWithdrawBlock = fromBlock + unlockPeriod + 1 // +1 to account for the block created executing unlock
```

Comment:
We can avoid this +1 by moving the `fromBlock` line after the unlock tx

Reply:
agreed.. for some reason that wasn't working out for me so I postponed it for later. Fixed as suggested.